src/main/scala/com/github/codelionx/dodo/actors/Worker.scala

```diff
@@ -59,33 +59,35 @@ class Worker(resultCollector: ActorRef) extends Actor with ActorLogging with Dep
       sender ! OrderEquivalent(oeToCheck, checkOrderEquivalent(table(oeToCheck._1), table(oeToCheck._2)))
       sender ! GetTask
 
-    case CheckForOD(odCandidate, reducedColumns) =>
-      val ocdCandidate = (odCandidate._1 ++ odCandidate._2, odCandidate._2 ++ odCandidate._1)
-      var foundOD = false
-      if (checkOrderDependent(ocdCandidate, table.asInstanceOf[Array[TypedColumn[_]]])) {
-
-        var newCandidates: Queue[(Seq[Int], Seq[Int])] = Queue.empty
-
-        if (checkOrderDependent(odCandidate, table.asInstanceOf[Array[TypedColumn[_]]])) {
-          resultCollector ! OD(substituteColumnNames(odCandidate, table))
-          foundOD = true
-        } else {
-          newCandidates ++= generateODCandidates(reducedColumns, odCandidate)
-        }
-        val mirroredOD = odCandidate.swap
-        if (checkOrderDependent(mirroredOD, table.asInstanceOf[Array[TypedColumn[_]]])) {
-          resultCollector ! OD(substituteColumnNames(mirroredOD, table))
-          foundOD = true
-        } else {
-          newCandidates ++= generateODCandidates(reducedColumns, odCandidate, leftSide = false)
-        }
-        sender ! ODsToCheck(odCandidate, newCandidates)
-        if (!foundOD || !settings.ocdComparability ) {
-          resultCollector ! OCD(substituteColumnNames(odCandidate, table))
+    case CheckForOD(odCandidates, reducedColumns) =>
+      var newCandidates: Queue[(Seq[Int], Seq[Int])] = Queue.empty
+      var foundODs: Seq[(Seq[String], Seq[String])] = Seq.empty
+      var foundOCDs: Seq[(Seq[String], Seq[String])] = Seq.empty
+      for (odCandidate <- odCandidates) {
+        val ocdCandidate = (odCandidate._1 ++ odCandidate._2, odCandidate._2 ++ odCandidate._1)
+        var foundOD = false
+        if (checkOrderDependent(ocdCandidate, table.asInstanceOf[Array[TypedColumn[_]]])) {
+          if (checkOrderDependent(odCandidate, table.asInstanceOf[Array[TypedColumn[_]]])) {
+            foundODs :+= substituteColumnNames(odCandidate, table)
+            foundOD = true
+          } else {
+            newCandidates ++= generateODCandidates(reducedColumns, odCandidate)
+          }
+          val mirroredOD = odCandidate.swap
+          if (checkOrderDependent(mirroredOD, table.asInstanceOf[Array[TypedColumn[_]]])) {
+            foundODs :+= substituteColumnNames(mirroredOD, table)
+            foundOD = true
+          } else {
+            newCandidates ++= generateODCandidates(reducedColumns, odCandidate, leftSide = false)
+          }
+          if (!foundOD || !settings.ocdComparability ) {
+            foundOCDs :+= substituteColumnNames(odCandidate, table)
+          }
         }
-      } else {
-        sender ! ODsToCheck(odCandidate, Queue.empty)
       }
+      resultCollector ! ODs(foundODs)
+      resultCollector ! OCDs(foundOCDs)
```

Comment:
Should we only send the OCDs if `foundOCDs.nonEmpty`?

We could also combine all results in one message `case class Result(ods: Seq[(Seq[String], Seq[String])], ocds: Seq[(Seq[String], Seq[String])]`